packages/plugins/experimental/plugin-inbox/src/components/Mailbox/Mailbox.tsx

```diff
@@ -17,67 +24,182 @@ const byDate =
   ({ created: a = '' }: MessageType, { created: b = '' }: MessageType) =>
     a < b ? -direction : a > b ? direction : 0;
 
-const setMessageProperty = (message: MessageType, property: string, value: any) => {
-  if (!message.properties) {
-    message.properties = {};
-  }
+const updateMessageProperty = (messages: MessageType[], property: string, value: any) => {
+  messages.forEach((message) => {
+    if (!message.properties) {
+      message.properties = {};
+    }
 
-  message.properties[property] = value;
+    message.properties[property] = value;
+  });
 };
 
 export type MailboxOptions = { readTimout?: number };
 
+const messageRowDefault = {
+  grid: { size: 56 },
+};
+
+const messageColumnDefault = {
+  grid: { size: 100 },
+};
+
+const hashString = (str?: string): number => {
+  if (!str) {
+    return 0;
+  }
+  return Math.abs(str.split('').reduce((hash, char) => (hash << 5) + hash + char.charCodeAt(0), 0));
+};
+
+const renderMessageCell = (message: MessageType, now: Date) => {
+  const text = message.blocks.find((block) => block.type === 'text')?.text;
+  const date = formatDate(now, message.created ? new Date(message.created) : new Date());
+  const from = message.sender?.name ?? message.sender?.email;
+  const subject = message.properties?.subject ?? text;
+  const hue = toHue(hashString(from));
+
+  return `<button class="message__thumb dx-focus-ring-inset" data-inbox-action="select-message" data-message-id="${message.id}"
+    ><dx-avatar hue=${hue} variant="square" hueVariant="surface" fallback="${from ? getFirstTwoRenderableChars(from).join('') : '?'}"></dx-avatar
+  ></button><button class="message__abstract dx-focus-ring-inset" data-inbox-action="current-message" data-message-id="${message.id}"
+    ><p class="message__abstract__heading" role="none"><span>${from}</span><span>${date}</span></p
+    ><p class="message__abstract__body">${subject}</p
+  ></button>`;
+};
+
+const messageCellClassName = 'message';
+
 // TODO(burdon): Extract contacts/orgs.
 // TODO(burdon): Split message body into parts and allow trim (e.g., remove forwarded part). Message as light stack?
 // TODO(burdon): Highlight message/chunks for action (e.g., follow-up, triggers embedding).
 // TODO(burdon): Create outline/kanban.
 // TODO(burdon): Address book/cards.
 
-export type MailboxProps = { mailbox: MailboxType; options?: MailboxOptions };
+export type MailboxProps = { mailbox: MailboxType; options?: MailboxOptions; ignoreAttention?: boolean };
 
-export const Mailbox = ({ mailbox, options = {} }: MailboxProps) => {
-  const [selected, setSelected] = useState<string>();
+export const Mailbox = ({ mailbox, options = {}, ignoreAttention }: MailboxProps) => {
+  const [currentMessageId, setCurrentMessageId] = useState<string>();
   const queue = useQueue<MessageType>(mailbox.queue.dxn, { pollInterval: 1_000 });
   const tRef = useRef<ReturnType<typeof setTimeout>>();
+  const { hasAttention } = useAttention(mailbox.id);
```

Comment:
should this be fully qualified id?

Reply:
AFAIK attention does not use fully qualified IDs because the original objects are not always available (or relevant).

Reply:
I think I’m wrong on this, seeing as SheetContainer uses fullyQualified Id, will fix here.